Documentation/ModuleAliasing.md

```diff
@@ -0,0 +1,102 @@
+# Module Aliasing by SwiftPM
+
+## Overview
+
+The number of package dependencies often grows, with that, a name collision can occur among modules from different packages. Module names such as `Logging` or `Utils` are common examples. In order to resolve the collision, SwiftPM (in 5.7+) introduces a new parameter `moduleAliases`, which allows a user to define new unique names for the conflicting modules without requiring any source code changes.  
+
+## How to Use
+
+Consider the following scenario. `App` imports a module called `Game` from package `swift-game`, and a module called `Utils` from package `swift-draw`. 
+
+```
+App
+  |— Module Game (from package ‘swift-game’)
+  |— Module Utils (from package ‘swift-draw’)
+```
+
+The `App` package manifest has the following product dependencies.
+
+```
+ targets: [
+  .executableTarget(
+    name: "App",
+    dependencies: [
+     .product(name: "Game", package: "swift-game"),
+     .product(name: "Utils", package: "swift-draw"),
+   ])
+ ]
+```
+
+There's no collision so far, so everything builds fine. Now `App` updates the version of the `swift-game` package to the latest, and in that version, the package contains a module called `Utils` as one of target dependencies for `Game`. This conflicts with module `Utils` from package `swift-draw`.
+
+```
+App
+  |— Module Game (from package ‘swift-game’)
+      |— Module Utils (from package ‘swift-game’)
+  |— Module Utils (from package ‘swift-draw’)
+```
+
+To resolve the collision, we can now use a new parameter `moduleAliases` as follows.
+
+```
+ targets: [
+  .executableTarget(
+    name: "App",
+    dependencies: [
+     .product(name: "Game",
+              package: "swift-game",
+              moduleAliases: ["Utils": "GameUtils"]),
+     .product(name: "Utils",
+              package: "swift-draw"),
+   ])
+ ]
+```
+
+This will rename the `Utils` module in packgae `swift-game` to be a new user-provided name `GameUtils` and compile all the source references to `Utils` as `GameUtils` in targets of the product in the package. The name of the binary for the `Utils` module will be `GameUtils.swiftmodule`. No source changes are required by the `swift-game` package. 
+
+If `App` wants to import the `Utils` module from `swift-game`, it needs to directly reference the aliased module in its sosurce code, e.g. `import GameUtils`. If it already contains `import Utils`, it will continue to refer to the `Utils` module from package `swift-draw`, as expected.
```

Comment:
```suggestion
If `App` wants to import the `Utils` module from `swift-game`, it needs to directly reference the aliased module in its source code, e.g. `import GameUtils`. If it already contains `import Utils`, it will continue to refer to the `Utils` module from package `swift-draw`, as expected.
```